### PR TITLE
Cosmosdb tasks

### PR DESCRIPTION
**Pull request by frsann**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
Add Azure Cosmos DB tasks.

## Why is this PR important?
Helps Azure Cosmos DB users create and query data as part of their Flows.

### CODE REVIEW

File: src/prefect/tasks/azure/cosmosdb.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+        """
+
+        if url is None:
+            raise ValueError("A url must be provided.")
+
+        if database_or_container_link is None:
+            raise ValueError("A database or container link must be provided.")
+
+        if item is None:
+            raise ValueError("A item must be provided.")
```
Comment (joshmeek) [inline src/prefect/tasks/azure/cosmosdb.py]:
```suggestion
            raise ValueError("An item must be provided.")
```

Comment (frsann) [reply src/prefect/tasks/azure/cosmosdb.py]: Grammar is hard. Fixed.

File: src/prefect/tasks/azure/cosmosdb.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+                that stores your Azure credentials; this Secret must be JSON string with the key
+                `AZ_COSMOS_AUTH`. The value should be dictionary containing `masterKey` or `resourceTokens`, 
+                where the `masterKey` value is the default authorization key to use to
+                create the client, and `resourceTokens` value is the alternative
+                authorization key.
+            - options (dict, optional): options to be passed to the 
+                `azure.cosmos.cosmos_client.CosmosClient.ReadItem` or `ReadItems` method.
+
+        Returns:
+            - (dict or azure.cosmos.query_iterable.QueryIterable)): a single document or all documents.
```
Comment (cicdw) [inline src/prefect/tasks/azure/cosmosdb.py]: Is the `QueryIterable` object an iterator or a fully hydrated list?  Ultimately I'm wondering if you happen to know whether it's `cloudpickle`able.

File: src/prefect/tasks/azure/cosmosdb.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+                `AZ_COSMOS_AUTH`. The value should be dictionary containing `masterKey` or `resourceTokens`, 
+                where the `masterKey` value is the default authorization key to use to
+                create the client, and `resourceTokens` value is the alternative
+                authorization key.
+            - options (dict, optional): options to be passed to the 
+                `azure.cosmos.cosmos_client.CosmosClient.QueryItems` method.
+            - partition_key (str, None): Partition key for the query.
+
+        Returns:
+            - (azure.cosmos.query_iterable.QueryIterable)): the query result
```
Comment (cicdw) [inline src/prefect/tasks/azure/cosmosdb.py]: same question as above

File: src/prefect/tasks/azure/cosmosdb.py (hunk 2 above)
Comment (frsann) [reply src/prefect/tasks/azure/cosmosdb.py]: Good catch! I changed `CosmosDBReadItems` and `CosmosDBQueryItems` to return lists, following the example in `prefect.tasks.google.bigquery.BigQueryTask`

File: src/prefect/tasks/azure/cosmosdb.py (hunk 3 above)
Comment (frsann) [reply src/prefect/tasks/azure/cosmosdb.py]: Answered above.

**Pull request merged**